### ISSUE TITLE
no response on second peer connection stun request 

### ISSUE THREAD

**Issue by koush** [shinyoshiaki/werift-webrtc#217]
I've been having a hard time tracking this issue down, and was wanting to see if you had any insight:

Clients:
Offer: Safari iOS (on cellular)
Answer: Werift on Mac

On first run of connecting the peers, werift will successfully retrieve the srflx candidates from the STUN server. On second or third run, werift will fail to retrieve a response from the STUN server, causing the connection to fail. After a few minutes, the STUN server seems to start responding again.

I have observed this behavior with multiple different stun servers (with no quotas enabled). I only observe this behavior when the peer is on cellular. When the Safari iOS peer is on local network, Werift will successfully receive a STUN response every time.

I'm unsure why the cellular network peer would cause STUN requests to fail in werift. I verified that the requests are being sent, and no response is received on the datagram socket.

Do you have any ideas?

My guess is that werift is somehow sending bad stun requests after the first one.

**Comment by koush** [comment]
I figured this out, I think. The iOS device is trickling candidates on cellular, some of which end with `.local`, which will be unreachable by the Mac since it's not on the local network.

These `.local` hostnames end up being resolved here:

https://github.com/shinyoshiaki/werift-webrtc/blob/629df812e07131cd2b5af8608e2268c8259b2752/packages/ice/src/ice.ts#L448

The issue seems to be that these local lookups take a long time to timeout, and they are run by the operating system or the node.js process serially. Meaning, only 1 lookup happens at a time. So, when the request to lookup stun.l.google.com comes in, it has to wait in line behind all the other `.local` lookups that are pending. That is why:

> After a few minutes, the STUN server seems to start responding again.

I'm not entirely sure this is the case, but I verified that forcing `.local` addresses to fail (or fall through to the non-lookup path) allows the stun requests to complete. Something with Mac dns.lookup is causing a udp send with an unresolved host string to fail.

> dns.lookup() does not necessarily have anything to do with the DNS protocol. The implementation uses an operating system facility that can associate names with addresses, and vice versa. This implementation can have subtle but important consequences on the behavior of any Node.js program. Please take some time to consult the [Implementation considerations section](https://nodejs.org/api/dns.html#implementation-considerations) before using dns.lookup().
